### PR TITLE
Fix #192, Adds check commit message job in format check

### LINKED ISSUE #192: Check that commit messages adhere to standard using the continuous integration workflow
**Is your feature request related to a problem? Please describe.**
Community contributors are not familiar with our `Fix #XYZ, SHORT_DESCRIPTION` convention for commit messages. 

**Describe the solution you'd like**
Adding an extra step or job to "format check" workflow

**Describe alternatives you've considered**
Add a pre-commit hook on git

**Additional context**
None

**Requester Info**
@astrogeco

### PR DESCRIPTION
**Pull request by chillfig**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
A clear and concise description of what the contribution is.
- Fixes #192 

**Testing performed**
Tested on fork: https://github.com/chillfig/cFS/runs/6427976618?check_suite_focus=true

**Expected behavior changes**
Commit messages not following any of the below conventions will fail the [format-check.yml
](https://github.com/nasa/cFS/blob/main/.github/workflows/format-check.yml)
* `Fix #XYZ, DESCRIPTION` 
* `HotFix #XYZ, DESCRIPTION` 
* `Partial #XYZ, DESCRIPTION` 
* `Merge pull request #XYZ DESCRIPTION` 
* `IC: DESCRIPTION` 

**System(s) tested on**
Ubuntu 18.04

**Additional context**
The regex for `DESCRIPTION` is defined as a string of 1 or more alphanumeric characters.
The regex for `XYZ` is defined as a string of 1 or more numeric characters.

**Code contributions**
The cFS repository is provided to bundle the cFS Framework.  It is utilized for bundling submodules, continuous integration testing, and version management and does not contain any software.  Code contributions should be directed to the appropriate submodule.

**Contributor Info - All information REQUIRED for consideration of pull request**
Justin Figueroa, ASRC Federal

### CODE REVIEW

**Comment by astrogeco** [comment]
CCB:2022-05-04, APPROVED with CHANGES
- Spin off into its own reusable workflow (triggered on `workflow_call`)
- Consider replacing "partial" with "part"? Do we have a "better" short word for that?

**Comment by skliper** [comment]
@astrogeco are you able to separate it into it's own job that only runs on pull requests even when triggered from a workflow call?  That would be enough to separate it from the code format check (different job) and avoid nuisance failures (only on pull request) for me.  Although if there's plans to add more checking to this workflow a name update may help clarify things (not just checking code whitespace formatting anymore).

**Comment by chillfig** [comment]
> @astrogeco are you able to separate it into it's own job that only runs on pull requests even when triggered from a workflow call? That would be enough to separate it from the code format check (different job) and avoid nuisance failures (only on pull request) for me. Although if there's plans to add more checking to this workflow a name update may help clarify things (not just checking code whitespace formatting anymore).

Interesting. Yes, the "Check commit message" step can be changed into its own job in [format-check.yml](https://github.com/nasa/cFS/blob/main/.github/workflows/format-check.yml) with the addition of `if: github.event_name == 'pull_request'`. According to https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif, that addition should keep the check commit message job to only running on pull requests.

File: .github/workflows/format-check.yml [hunk 1]
```diff
@@ -65,3 +65,17 @@ jobs:
...
+    # Only run for pull-requests.
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-18.04
+    timeout-minutes: 15
+    steps:
+      - name: Check for Fix / HotFix / Partial
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^((Fix|HotFix|Partial)\s\#[0-9]+,\s[a-zA-Z0-9]+|Merge\spull\srequest\s\#[0-9]+\s[a-zA-Z0-9]+|IC:\s[a-zA-Z0-9]+)'
+          error: 'You need at least one "Fix #<issue number>, <short description>" line. "HotFix" and "Partial" are also acceptable.'
```
Comment (skliper) [inline .github/workflows/format-check.yml:81]: How about "Part" vs "Partial"?

**Pull request merged**